app/src/main/java/com/keylesspalace/tusky/network/InstanceSwitchAuthInterceptor.java

```diff
@@ -55,13 +54,26 @@ public Response intercept(@NonNull Chain chain) throws IOException {
                 builder.url(swapHost(originalRequest.url(), instanceHeader));
                 builder.removeHeader(MastodonApi.DOMAIN_HEADER);
             } else if (currentAccount != null) {
-                //use domain of current account
-                builder.url(swapHost(originalRequest.url(), currentAccount.getDomain()))
-                        .header("Authorization",
-                                String.format("Bearer %s", currentAccount.getAccessToken()));
+                String accessToken = currentAccount.getAccessToken();
+                if (!accessToken.isEmpty()) {
+                    //use domain of current account
+                    builder.url(swapHost(originalRequest.url(), currentAccount.getDomain()))
```

Comment:
I understand that we can only add access token when we have it but what about switching domains? Should it also depend on token or just on account?

Reply:
I think it is fine like that, this branch really is only for authenticated requests to the current instance. For everything else one would use the DOMAIN_HEADER.